xdsl/interactive/transforms/experimental/individual_rewrite_interactive.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.interactive.transforms.canonicalization_patterns.arith import (
+    arith_op_to_rewrite_pattern,
+)
+from xdsl.ir import MLContext
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.tools.command_line_tool import get_all_dialects
+from xdsl.transforms import individual_rewrite
+
+INTERACTIVE_REWRITE_BY_NAMES: dict[str, dict[str, RewritePattern]] = {
+    op.name: {
+        pattern.__class__.__name__: pattern
+        for pattern in arith_op_to_rewrite_pattern[op]
+    }
+    for dialect in get_all_dialects().values()
+    for op in dialect().operations
+    if op in arith_op_to_rewrite_pattern
+}
+"""
+Returns a dictionary representing all possible experimental interactive rewrites. Keys are operation names, and
+values are dictionaries. In the inner dictionary, the keys are names of patterns
+associated with each operation, and the values are the corresponding RewritePattern
+instances.
+"""
+
+ALL_REWRITES: dict[str, dict[str, RewritePattern]] = {
+    op_name: {
+        **individual_rewrite.REWRITE_BY_NAMES.get(op_name, {}),
+        **INTERACTIVE_REWRITE_BY_NAMES.get(op_name, {}),
+    }
+    for op_name in set(individual_rewrite.REWRITE_BY_NAMES)
+    | set(INTERACTIVE_REWRITE_BY_NAMES)
+}
+"""
+Concatenates REWRITE_BY_NAMES and INTERACTIVE_REWRITE_BY_NAMES.
+"""
+
+
+@dataclass
+class IndividualRewriteInteractive(individual_rewrite.IndividualRewrite):
```

Comment:
It feels a little off to subclass this rewrite that's already pretty specific to the interactive compilation stack, we might as well add the custom arith patterns to the dictionary it uses directly